### ISSUE TITLE
Fails to validate string enum across modules

### ISSUE THREAD

**Issue by Skinney** [facebook/flow#1718]
In one file, `types/stakeholder.js` I have this type:

```
export type StakeholderType = 'PERSON' | 'COMPANY' | 'EMPLOYEE';
```

This is then imported into `components/stakeholder-adder.jsx`, where it is used for the props of the React component.

```
type Props = {
   typeFilter: Array<StakeholderType>
};
```

Then, in another file `components/stakeholder-modal.jsx` I call the stakeholder component thusly:

```
<StakeholderAdder
    typeFilter={['PERSON']}/>
```

This fails to validate with the message: string is incompatible with StakeholderType.

It seems that flow forgets that StakeholderType is a string enum when validating over several modules.

**Comment by avikchaudhuri** [comment]
Following instructions, could not repro with 0.26.0. 

Found 0 errors.

In `components/stakeholder-modal.jsx` I changed 'PERSON' to 'PERSON#' and get an expected error, confirming that my repro wasn't passing by accident.

**Comment by Skinney** [comment]
Wierd. I still have the same problem:

```bash
➜  Jarvis git:(master) flow version
Flow, a static type checker for JavaScript, version 0.26.0
➜  Jarvis git:(master) flow
client/src/js/views/order/stakeholders/company.jsx:119
119:                 <StakeholderAdder
                     ^ React element `StakeholderAdder`
121:                     typeFilter={['PERSON']}
                                      ^^^^^^^^ string. This type is incompatible with
 14:     typeFilter: ?Array<StakeholderType>
                            ^^^^^^^^^^^^^^^ StakeholderType. See: client/src/js/views/shared/forms/stakeholder-adder.jsx:14

client/src/js/views/order/stakeholders/person.jsx:112
112:                 <StakeholderAdder
                     ^ React element `StakeholderAdder`
114:                     typeFilter={['COMPANY']}
                                      ^^^^^^^^^ string. This type is incompatible with
 14:     typeFilter: ?Array<StakeholderType>
                            ^^^^^^^^^^^^^^^ StakeholderType. See: client/src/js/views/shared/forms/stakeholder-adder.jsx:14


Found 2 errors
```

**Comment by Skinney** [comment]
I'll try to double check the minimum test case later today, and come up with a new one if I can't reproduce.

**Comment by Skinney** [comment]
Fixed in 0.27.0
